spec/views/vacancies/show.html.slim_spec.rb

```diff
@@ -1,11 +1,81 @@
 require "rails_helper"
 
 RSpec.describe "vacancies/show" do
+  let(:current_jobseeker) { nil }
+
   before do
+    allow(view).to receive(:current_jobseeker).and_return(current_jobseeker)
     assign :vacancy, VacancyPresenter.new(vacancy)
     render
   end
 
+  describe "How to apply button" do
+    context "when vacancy expired" do
+      let(:vacancy) { create(:vacancy, :expired, enable_job_applications: true) }
+
+      it "renders expiration notification" do
+        expect(rendered).to have_content(I18n.t("jobs.expired_listing.notification"))
+        expect(rendered).to have_no_link(I18n.t("jobseekers.job_applications.apply.apply"))
+      end
+    end
+
+    context "when applying with TV" do
+      let(:vacancy) { create(:vacancy, enable_job_applications: true) }
+
+      it "renders apply online details" do
+        expect(rendered).to have_content(I18n.t("jobseekers.job_applications.applying_for_the_job_paragraph"))
+        expect(rendered).to have_link(I18n.t("jobseekers.job_applications.apply.apply"))
+      end
+    end
+
+    context "when applying with new upload/download application form on TV" do
```

Comment:
I don't think this is part of this ticket. Isn't this ticket just about changing button text and opening the link in a new tab if the school wants users to apply on an external website?